src/app/pages/payment/pin-payment.component.ts

```diff
@@ -0,0 +1,86 @@
+import { validateInput } from './../../validators/payment-validators';
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormControl, Validators } from '@angular/forms'
+import { PaymentService } from 'src/app/services/payment-service';
+import { NgxSpinnerService } from "ngx-spinner";
+import { Router } from '@angular/router';
+import { ToastrService } from 'ngx-toastr';
+
+@Component({
+  selector: 'app-pin-payment',
+  templateUrl: './pin-payment.component.html',
+  styleUrls: ['./payment.component.scss']
+})
+export class PinPaymentComponent implements OnInit {
+	validYears: number[];
+	validMonths: string[] = [
+		'01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11', '12'
+	]
+	cardForm = new FormGroup({
+    cardNumber: new FormControl('5399838383838381', Validators.required),
+		securityNumber: new FormControl('564', Validators.required),
+		cardExpiry: new FormGroup({
+			expiryMonth: new FormControl(
+				'MM', [Validators.required, validateInput('MM')]
+				),
+			expiryYear: new FormControl(
+				'YYYY', [Validators.required, validateInput('YYYY')]
+				)
+		}),
+		amount: new FormControl('100', Validators.required),
+		pin: new FormControl('3310', Validators.required),
+		purpose: new FormControl('Saving', Validators.required),
+		saveCard: new FormControl(false)
+  });
+
+  constructor(
+		private paymentService: PaymentService, 
+		private router: Router,
+		private spinner: NgxSpinnerService,
+		private toastr: ToastrService,
+		) { }
+
+  ngOnInit() {
+		this.generateValidYears()
+	}
+	generateValidYears(){
+		let yearList: number[] = [];
+		const now = new Date();
+		const currentYear = now.getFullYear();
+		const lastYear = currentYear + 10;
+		for(let i = currentYear; i <= lastYear; i++){
+			yearList.push(i);
+		}
+		
+		this.validYears = yearList;
+	}
+	submitDetails() {
+		const payload = {
+			'cardno': this.cardForm.value['cardNumber'],
+			'cvv': this.cardForm.value['securityNumber'],
+			'expirymonth': this.cardForm.value['cardExpiry']['expiryMonth'],
+			'expiryyear': this.cardForm.value['cardExpiry']['expiryYear'].slice(-2),
+			'amount': this.cardForm.value['amount'],
+			'pin': this.cardForm.value['pin'],
+			'purpose': this.cardForm.value['purpose'],
+			'save_card': this.cardForm.value['saveCard']
+		}
+		this.spinner.show()
+		this.paymentService.initiatePinPay(payload).subscribe(
+			resp => {
+				this.spinner.hide()
+				this.router.navigate(['/validate-pin', resp['flwRef'], 
+				this.cardForm.value['purpose']])
+			},
+			error => {
+				this.spinner.hide()
+				this.toastr.error(error['error']['message'], 'Error')
+			}
+
+		)
+	}
+	onBack():void {
+	this.router.navigate(['/home']);
```

Comment:
Hey @Victoradukwu, regarding the back navigation going to the homepage, sometimes, the user could previously have come from for example the profile page and at that point, clicking the back button should take them back to the location they last visited.